lib/parachain/types/types_test.go

```diff
@@ -371,3 +371,46 @@ func TestMustHexTo32BArray(t *testing.T) {
 	result := mustHexTo32BArray(t, inputHex)
 	require.Equal(t, expectedArray, result)
 }
+
+func TestPersistedValidationData(t *testing.T) {
+	expected := []byte{12, 7, 8, 9, 10, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 4, 0, 0} //nolint:lll
+
+	pvd := PersistedValidationData{
+		ParentHead:             HeadData{Data: []byte{7, 8, 9}},
+		RelayParentNumber:      uint32(10),
+		RelayParentStorageRoot: common.Hash{},
+		MaxPovSize:             uint32(1024),
+	}
+
+	actual, err := scale.Marshal(pvd)
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+
+	newpvd := PersistedValidationData{}
+	err = scale.Unmarshal(actual, &newpvd)
+	require.NoError(t, err)
+	require.Equal(t, pvd, newpvd)
+}
+
+func TestOccupiedCoreAssumption(t *testing.T) {
```

Comment:
also test for unmarshal.
you can refer [TestCollationProtocol](https://github.com/ChainSafe/gossamer/blob/cae09d69e2c388851ed92d1aa3eedcef1dadebe9/lib/parachain/collation_protocol_test.go#L29-L147)